crates/contract/src/call.rs

```diff
@@ -131,6 +131,20 @@ pub struct CallBuilder<T, P, D, N: Network = Ethereum> {
     transport: PhantomData<T>,
 }
 
+impl<T, P, D, N> CallBuilder<T, P, D, N>
+where
+    T: Transport + Clone,
+    P: Provider<T, N>,
+    D: CallDecoder + Send + Sync + Unpin,
+    N: Network,
+    Self: 'static,
+{
+    /// Retrieves a reference to the transaction request.
+    pub fn request(&self) -> &N::TransactionRequest {
```

Comment:
`impl AsRef<N::TransactionRequest>`

Reply:
> `impl AsRef<N::TransactionRequest>`

sorry for the question but you want me to return impl AsRef<N::TransactionRequest> or you want me to implement impl AsRef<N::TransactionRequest> for CallBuilder ? so we can use as_ref ?

Reply:
i assum we want to return impl AsRef<N::TransactionRequest>